### DISCUSSION TITLE
Problem with Rmanager running knitr

### DISCUSSION THREAD

**Discussion by gafox1** [WinEdt-Team/WinEdt-Support#71] [1.2 Bugs and Issues]
Rmanager is failing to execute knitr or purl on my Windows 11 machine. For example, trying to run knitr + PDFLaTeX on a .Rnw script, I get this sort of message:

Command Line:   "C:\Program Files (x86)\R\R-4.4.1\bin\x64\Rscript.exe" -e "C:\Users\gafox\AppData\Roaming\WinEdt Team\WinEdt 11\RManager\Exec\Knitr.R" knitr::knit2pdf('SemaphoreSupplement-Final.Rnw')
Startup Folder: C:\Users\gafox\Projects\SemaphoreCactus

Error: unexpected symbol in "C:\Users"
Execution halted

I'm baffled as to where to even look for the problem. I should add that all the components work: using the Rmanager menus in WinEdt, I can open Rgui, knit from the R command line, and then open the .tex file, and compile it with PDFLaTeX from the menu. 

An obvious suspect is the space in the path for the Knitr.R file, but this hasn't previously been a problem. And that path was set by the WinEdt installer, not by me, so I don't think this is it.

Any suggestions on how to fix this are welcome!

**Comment by Alex-Simonic** [comment]
The command line is not quite right. Switch -e cannot be used like this. Start Options -> Execution Modes and click on RScript. Switches should be empty and Parameters should be `"%N%T"`. You must have changed that to `-e` and `knitr::knit2pdf('%N%T')`, respectively. That will not work because the default RManager macros insert `"%b\RManager\Exec\Knitr.R"` in the command line before Parameters and this is not compatible with using the `-e` switch.

You will have to either use the default parameters for RScript or else you will have to modify three macros that insert `"%b\RManager\Exec\Knitr.R"` in the command line.

I did not prepare these macros but it has been like this for a long time...

**Reply by gafox1** [reply]
That works. Thanks so much!
Gordon
